Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/signinsignup/SignInScreen.kt

```diff
@@ -122,46 +145,44 @@ fun SignInContent(
     }
 }
 
+@OptIn(ExperimentalMaterialApi::class)
 @Composable
 fun ErrorSnackbar(
-    showError: Boolean,
-    errorText: String,
+    snackbarHostState: SnackbarHostState,
     modifier: Modifier = Modifier,
     onDismiss: () -> Unit = { }
 ) {
-    if (!showError) {
-        return
-    }
-
-    // Make Snackbar disappear after 5 seconds if the user hasn't interacted with it
-    launchInComposition() {
-        delay(5000L)
-        onDismiss()
-    }
-
-    Box(modifier = modifier.fillMaxWidth().wrapContentHeight(Alignment.Bottom)) {
-        Crossfade(current = showError) {
-            Snackbar(
-                modifier = Modifier.padding(16.dp),
-                text = {
-                    Text(
-                        text = errorText,
-                        style = MaterialTheme.typography.body2
-                    )
-                },
-                action = {
-                    TextButton(
-                        onClick = onDismiss,
-                        contentColor = contentColor()
-                    ) {
+    snackbarHostState.currentSnackbarData?.let { data ->
```

Comment:
Do you need this? The `snackbarHostState` should handle this for you.

Reply:
`currentSnackbarData` is nullable so when I try to set the `message` I'd end up with a nullable string. 
I could do `snackbarHostState.currentSnackbarData?.message?.let { ... }` in the snackbar text lambda, but what's the point of showing a snackbar without a message?

Reply:
Ahhh I see. Yeah remove this, and then use the `data` pass into the `snackbar` lambda below